core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
```

Comment:
There may be discrepancies between pure Java and Scala behavior, but "fields" from IR includes parameters to the primary constructor since they do turn into a field sometimes.
The following is using 0.10.x:

```scala
scala> class Foo(x0: Int) { val x = x0 }
defined class Foo

scala> new Foo(1).pickle
res0: scala.pickling.json.pickleFormat.PickleType =
JSONPickle({
  "$type": "Foo",
  "x": 1
})

scala> class Bar(x0: Int) { val x = x0; override def toString = x0.toString }
defined class Bar

scala> new Bar(1).pickle
res1: scala.pickling.json.pickleFormat.PickleType =
JSONPickle({
  "$type": "Bar",
  "x": 1,
  "x0": 1
})
```

My change would no longer attempt to capture `x0`. It's currently using weird try-catch-em-all, and I figured that's the source of the problem. I think I need to check if the given type is pure Java or not.